### PR TITLE
feat: easy assign

### PR DESCRIPTION
**Pull request by juniusfree**
## :pushpin: References

- **Wonder issue:** https://app.wonderverse.xyz/organization/wonderverse/boards?task=70128747248878282&entity=task
- **Related pull-requests:** https://github.com/wondrous-dev/wondrous-backend/pull/618

## :blue_book: Description

Assign reviewers or assignee from the task view modal

## :movie_camera: Screenshot or Video

[easy-assign.webm](https://user-images.githubusercontent.com/8164667/197149463-c624007a-54fc-4e58-b75c-6ba3d6cdf88b.webm)

## :cake: Checklist:

- [x] I self-reviewed my changes
- [x] My changes follow the style guide: https://creative-earth-33e.notion.site/TT-Wonder-Style-guide-4702a45133374148953bfcaf584120b7
- [x] I tested my changes in Chrome

### CODE REVIEW

File: wondrous-app/components/Common/TaskViewModal/styles.tsx [hunk 1]
```diff
@@ -887,3 +891,34 @@ export const TaskIntiativesContainer = styled.div`
...
+      width: 26px;
+      height: 26px;
+      &:hover {
+        background: ${palette.grey920};
+      }
+    }
+  }
+`;
+
+export const ReviewerWrapper = styled.div`
```
Comment (vefimov) [inline wondrous-app/components/Common/TaskViewModal/styles.tsx:914]: You can use `Grid` from mui

File: wondrous-app/components/Common/TaskViewModal/styles.tsx [hunk 2]
```diff
@@ -887,3 +891,34 @@ export const TaskIntiativesContainer = styled.div`
...
+  width: 100%;
+  flex-wrap: wrap;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  gap: 12px;
+  > div:nth-last-child(2) {
```
Comment (vefimov) [inline wondrous-app/components/Common/TaskViewModal/styles.tsx:921]: Better and more safe to add a CSS class to the element

File: wondrous-app/components/Common/TaskViewModal/taskViewModalFields.tsx [hunk 3]
```diff
@@ -1,75 +1,149 @@
...
+    withTaskReviewers < eligibleReviewers?.length &&
+    !isEmpty(filteredEligibleReviewers);
+  const selfReviewer = !taskReviewerIds?.includes(user.id) && user;
 
   if (!shouldDisplay) {
     return null;
   }
   return (
-    <TaskSectionDisplayDiv>
+    <TaskSectionDisplayDiv style={{ alignItems: 'start' }}>
```
Comment (vefimov) [inline wondrous-app/components/Common/TaskViewModal/taskViewModalFields.tsx:96]: Add style directly to the `TaskSectionDisplayDiv`

File: wondrous-app/components/Common/TaskViewModal/taskViewModalFields.tsx [hunk 4]
```diff
@@ -1,75 +1,149 @@
...
+                      handleRemove={(e) => {
+                        e.stopPropagation();
+                        handleUpdateReviewers(taskReviewerIds.filter((id) => id !== taskReviewer.id));
+                      }}
+                      canEdit={canEdit}
+                    />
+                  )}
+                  onClick={() => {
+                    handleClose();
+                    router.push(`/profile/${taskReviewer?.username}/about`, undefined, {
```
Comment (vefimov) [inline wondrous-app/components/Common/TaskViewModal/taskViewModalFields.tsx:117]: Move this code to the `handleClose` event handler 
```
           router.push(`/profile/${taskReviewer?.username}/about`, undefined, {
                      shallow: true,
                    });
```

**Review by vefimov — CHANGES_REQUESTED** [review]

File: wondrous-app/components/Common/TaskViewModal/styles.tsx (hunk 2 above)
Comment (juniusfree) [reply wondrous-app/components/Common/TaskViewModal/styles.tsx:921]: Why?

File: wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+const ProfilePicture = ({ profilePicture }) =>
+  profilePicture ? (
+    <SafeImage useNextImage={false} src={profilePicture} width={24} height={24} style={{ borderRadius: '24px' }} />
+  ) : (
+    <DefaultUserImage width={24} height={24} />
+  );
+
+const RenderOption = (props, option) => {
+  const { profilePicture, label, hide } = option || {};
+  if (hide) return null;
```
[the remark this answers is not in the document]
Comment (juniusfree) [reply wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx:31]: Why?

File: wondrous-app/components/Common/TaskViewModal/styles.tsx (hunk 2 above)
Comment (Lamperoyge) [reply wondrous-app/components/Common/TaskViewModal/styles.tsx:921]: I agree with adding a class. The reason for this is at some point in the future if we modify the html structure we might miss changing it here

File: wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+import { ArrowWrapper, Option, PaperComponent, StyledAutocomplete, StyledTextField } from './styles';
+
+const profilePictureStyle = {
+  display: 'flex',
+  width: '24px',
+  height: '24px',
+  borderRadius: '50%',
+};
+
+const ProfilePicture = ({ profilePicture }) => (
```
Comment (Lamperoyge) [inline wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx:17]: we have a component components/Common/ProfilePictureHelpers can we reuse that here?

File: wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  if (hide) return null;
+  return (
+    <Option {...props}>
+      <ProfilePicture profilePicture={profilePicture} />
+      {label}
+    </Option>
+  );
+};
+
+const assignToSelf = ({ user, onClick }) => {
```
Comment (Lamperoyge) [inline wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx:40]: can we move the user check inside the component itself and pass user / onclick with props this way avoiding the assignToSelf function and simply having AssignToSelf component? imo more readable and I might be wrong but don't think we need a wrapper function here

**Review by Lamperoyge — CHANGES_REQUESTED** [review]

File: wondrous-app/components/Common/TaskViewModal/styles.tsx (hunk 2 above)
Comment (juniusfree) [reply wondrous-app/components/Common/TaskViewModal/styles.tsx:921]: How to set the class if the targeted element varies? On the reviewer section, some state updates the second-to-last element.

File: wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx (hunk 6 above)
Comment (juniusfree) [reply wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx:16]: `ProfilePictureHelpers` does not use `useNextImage`. @vefimov recommends using it.

**Comment by Lamperoyge** [comment]
One issue I discovered when testing is: clicking on the input when a user is assigned takes me to the user's page, I think it needs to open the dropdown and link to the user's profile only when clicking the user's profile pic / name ?

**Comment by juniusfree** [comment]
> One issue I discovered when testing is: clicking on the input when a user is assigned takes me to the user's page, I think it needs to open the dropdown and link to the user's profile only when clicking the user's profile pic / name ?

I'll ask Ben

**Comment by juniusfree** [comment]
@terryli0095 @Lamperoyge

reduced the clickable area to just the user info

https://user-images.githubusercontent.com/8164667/198213739-24acba13-1753-433e-9251-6d43d116219a.mp4

**Pull request merged**